test/cppsuite/test_harness/workload/database_model.h

```diff
@@ -90,60 +75,17 @@ class collection {
      * The set of keys should always be contiguous such that other threads calling get_key_count
      * will always know that the keys in existence are 0 -> _key_count - 1.
      */
-    uint64_t
+    void
     increase_key_count(uint64_t increment)
     {
-        std::lock_guard<std::mutex> lg(_mtx);
         _key_count += increment;
```

Comment:
Dont you need fetch_add here ?

Reply:
No I do not as we don't need the value we add to, additionally += is overloaded by std::atomic so its doing the right thing.